### PR TITLE
Make it possible to build/run tests on linux-arm64

### PR DESCRIPTION
**Pull request by omajid**
I wanted to run the roslyn test suite on Linux on arm64. Building everything worked, but running tests would fail with errors like:

    Errors Microsoft.CodeAnalysis.CSharp.Emit.UnitTests_1
    Could not find 'dotnet' host for the 'X64' architecture.

    You can resolve the problem by installing the 'X64' .NET.

With the changes in this commit, I can build and run all tests using:

    $ ./eng/build.sh --restore --build --pack
    $ ./eng/build.sh --test

I would, eventually, like to be able to run the tests on other platforms, including s390x and ppc64le. Hopefully that might help identify/fix issues like https://github.com/dotnet/roslyn/pull/74392 earlier.

One thing I am not too happy about in this change is how many places need an explicit addition of "linux-arm64". That won't scale as we want to add more architectures.

### CODE REVIEW

**Review by 333fred — APPROVED** [review]

**Comment by 333fred** [comment]
@jaredpar for a second review.

File: src/Tools/Source/RunTests/Options.cs [hunk 1]
```diff
@@ -138,10 +138,24 @@ internal class Options
...
+            System.Runtime.InteropServices.Architecture.Arm => "arm",
+            System.Runtime.InteropServices.Architecture.Arm64 => "arm64",
+            System.Runtime.InteropServices.Architecture.Armv6 => "armv6",
+            System.Runtime.InteropServices.Architecture.LoongArch64 => "loongarch64",
+            System.Runtime.InteropServices.Architecture.Ppc64le => "ppc64le",
+            System.Runtime.InteropServices.Architecture.S390x => "s390x",
+            System.Runtime.InteropServices.Architecture.Wasm => "wasm",
+            System.Runtime.InteropServices.Architecture.X86 => "x86",
+            System.Runtime.InteropServices.Architecture.X64 => "x64",
+            _ => throw new NotImplementedException()
```
Comment (jaredpar) [inline src/Tools/Source/RunTests/Options.cs:152]: This rough code path being copied three times bothers me a bit. Trying to think of a way to reduce it. 

This copy should be able to reference the one in `ILAsmUtilities.cs`. Or at least we should be able to put it in one location in that project and reference it here.

Comment (333fred) [reply src/Tools/Source/RunTests/Options.cs:152]: Yeah, it bothered me as well. My only thought had been putting it in a separate file and source-including it in each place, and I wasn't sure that was actually better.

File: src/Tools/Source/RunTests/Options.cs [hunk 2]
```diff
@@ -138,10 +138,24 @@ internal class Options
...
+        private static string GetArchitecture() => RuntimeInformation.ProcessArchitecture switch
+        {
+            System.Runtime.InteropServices.Architecture.Arm => "arm",
+            System.Runtime.InteropServices.Architecture.Arm64 => "arm64",
+            System.Runtime.InteropServices.Architecture.Armv6 => "armv6",
+            System.Runtime.InteropServices.Architecture.LoongArch64 => "loongarch64",
+            System.Runtime.InteropServices.Architecture.Ppc64le => "ppc64le",
+            System.Runtime.InteropServices.Architecture.S390x => "s390x",
+            System.Runtime.InteropServices.Architecture.Wasm => "wasm",
+            System.Runtime.InteropServices.Architecture.X86 => "x86",
```
Comment (omajid) [inline src/Tools/Source/RunTests/Options.cs:150]: I was thinking about this a bit: do we want to handle `x86` at all? IIRC there have been some problems in the past where this forced x86 in VisualStudio, even on x64 machines.

Comment (jaredpar) [reply src/Tools/Source/RunTests/Options.cs:150]: I would keep it there for completeness but this is only called on Linux paths and I don't think we really test x86 linux to my knowldege.

File: src/Tools/Source/RunTests/Options.cs (hunk 1 above)
Comment (jjonescz) [reply src/Tools/Source/RunTests/Options.cs:152]: I think we could simplify this whole method to `RuntimeInformation.ProcessArchitecture.ToString().ToLowerInvariant()`.

Comment (omajid) [reply src/Tools/Source/RunTests/Options.cs]: Done.

**Review by jaredpar — APPROVED** [review]

**Review by 333fred — APPROVED** [review]
@omajid, in the future please avoid force-pushing to PRs that are in-review. It prevents using a simple diff between the previously-reviewed commit and the current code.

**Pull request merged**